src/shared/test/integration/rpc-bridge-test.js

```diff
@@ -0,0 +1,185 @@
+      reciprocalBridge.call('connect');
+
+      await waitForMessageDelivery();
+      assert.equal(callbackCount, 3);
+    });
+  });
+
+  describe('sending and receiving RPC messages', () => {
+    it('should invoke Bridge method handler on every channel when calling a RPC method', async () => {
+      const bridge = createBridge();
+      const otherChannel = new MessageChannel();
+      bridge.createChannel(port1);
+      bridge.createChannel(otherChannel.port1);
+      const reciprocalBridge1 = createBridge();
+      const reciprocalBridge2 = createBridge();
+      reciprocalBridge1.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+      reciprocalBridge2.on('method1', (arg, cb) => cb(null, `${arg}bar`));
+      reciprocalBridge1.createChannel(port2);
+      reciprocalBridge2.createChannel(otherChannel.port2);
+
+      const results = await bridge.call('method1', 'param1');
+      assert.deepEqual(results.sort(), ['param1foo', 'param1bar'].sort());
+    });
+  });
+
+  describe('errors and timeouts', () => {
+    it(`raises an error when the listener's callback fails`, async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(errorMessage));
+      bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2);
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'param1');
+      } catch (err) {
+        error = err;
+      }
+      assert.equal(error, errorMessage);
+    });
+
+    it('destroys the Bridge channel when a RPC message fails', async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(errorMessage));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'destroy').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'param1');
+      } catch (err) {
+        error = err;
+      }
+
+      assert.called(channel.destroy);
+      assert.equal(error, errorMessage);
+      assert.deepEqual(reciprocalBridge.links, []);
+    });
+
+    it('no longer sends RPC messages to a Bridge channel that has received an error', async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(new Error(errorMessage)));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'call').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'param1');
+      } catch (err) {
+        error = err;
+      }
+      assert.equal(error.message, errorMessage);
+
+      for (let i = 0; i < 5; ++i) {
```

Comment:
Having 5 iterations here seems a bit redundant. Surely one would be enough?

Reply:
I removed the loop.